### ISSUE TITLE
Member profile update fails when no profession date is selected.

### ISSUE THREAD

**Issue by xmithd** [avengers5531/db_powon#43]
Expected behaviour:
A user could leave the profession and the dates blank (we have no constraint imposed)

Current behaviour when profession is left blank:
Fails with an uncaught exception
Environment: https://coc55311.encs.concordia.ca/
git checksum: 8a1446cdfe92abba158de842a42ee70c6c261a01
```
Details
Type: PDOException
Code: 22007
Message: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect date value: '' for column 'date_started' at row 1
File: /groups/c/co_comp5531_1/db_powon/src/Dao/Implementation/ProfessionDAOImpl.php
Line: 116
Trace

#0 /groups/c/co_comp5531_1/db_powon/src/Dao/Implementation/ProfessionDAOImpl.php(116): PDOStatement->execute()
#1 /groups/c/co_comp5531_1/db_powon/src/Services/Implementation/MemberServiceImpl.php(315): Powon\Dao\Implementation\ProfessionDAOImpl->updateProfessionForMember(Object(Powon\Entity\WorkAs))
#2 /groups/c/co_comp5531_1/db_powon/src/Routes/member_routes.php(124): Powon\Services\Implementation\MemberServiceImpl->updatePowonMember(Object(Powon\Entity\Member), Array)
#3 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#4 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(41): call_user_func(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#5 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/Route.php(325): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#6 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(116): Slim\Route->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#7 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/Route.php(297): Slim\Route->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#8 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/App.php(443): Slim\Route->run(Object(Slim\Http\Request), Object(Slim\Http\Response))
#9 /groups/c/co_comp5531_1/db_powon/src/Middleware/SessionLoader.php(47): Slim\App->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#10 [internal function]: Powon\Middleware\SessionLoader->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#11 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/DeferredCallable.php(37): call_user_func_array(Object(Powon\Middleware\SessionLoader), Array)
#12 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#13 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(67): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#14 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(116): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#15 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/App.php(337): Slim\App->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#16 /groups/c/co_comp5531_1/db_powon/vendor/slim/slim/Slim/App.php(298): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))
#17 /groups/c/co_comp5531_1/db_powon/public/index.php(19): Slim\App->run()
#18 {main}
```

How to reproduce:
Login as any user -> Update profile -> Choose profession Student and leave dates blank.
Click on update.

To summarize:
These are the things to fix:
1. Profession option should be allowed to be blank (without it failing of course)
2. Allow unspecified dates
3. Catch and log pdo exceptions at the Service level
